src/components/ChatPopUp.tsx

```diff
@@ -298,3 +287,62 @@ export function ChatPopUp(props: ChatPopUpProps) {
     </div>
   );
 }
+
+function usePanelState(
+  isOpen: boolean | undefined,
+  isOpenOnLoad: boolean | undefined,
+  initialMessageVisible: boolean | undefined
+) {
+  // control CSS behavior (fade-in/out animation) on open/close state of the panel.
+  const [showChat, setShowChat] = useState(false);
+  // control the actual DOM rendering of the panel. Start rendering on first open state
+  // to avoid message requests immediately on load while the popup is still "hidden"
+  const [renderChat, setRenderChat] = useState(false);
+  const [showInitialMessage, setshowInitialMessage] = useState(
+    initialMessageVisible
+  );
+
+  useEffect(() => {
+    if (isOpen !== undefined) {
+      setShowChat(isOpen);
+      setRenderChat(isOpen);
+    }
+  }, [renderChat, isOpen]);
```

Comment:
why is `renderChat` needed in deps list?

Reply:
oh that's weird, eslint didn't catch this. fixed